cluster/rpc/client/client.go

```diff
@@ -0,0 +1,245 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+
+	cmd "github.com/weaviate/weaviate/cluster/proto/api"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const serviceConfig = `
+{
+	"methodConfig": [
+		{
+			"name": [
+				{
+					"service": "weaviate.internal.cluster.ClusterService", "method": "Apply"
+				},
+				{
+					"service": "weaviate.internal.cluster.ClusterService", "method": "Query"
+				}
+			],
+			"waitForReady": true,
+			"retryPolicy": {
+				"MaxAttempts": 5,
+				"BackoffMultiplier": 2,
+				"InitialBackoff": "0.5s",
+				"MaxBackoff": "5s",
+				"RetryableStatusCodes": [
+					"ABORTED",
+					"RESOURCE_EXHAUSTED",
+					"INTERNAL",
+					"UNAVAILABLE",
+					"NOT_FOUND"
+				]
+			}
+		}
+	]
+}`
+
+type rpcAddressResolver interface {
+	// Address returns the RPC address corresponding to the given Raft address.
+	Address(raftAddress string) (string, error)
+}
+
+// Client is used for communication with remote nodes in a RAFT cluster
+// It wraps the gRPC client to our gRPC server that is running on the raft port on each node
+type Client struct {
+	rpc rpcAddressResolver
```

Comment:
**nitpick (non-blocking):** shall we call it `addrResolver` instead of `rpc` , wdyt?